rules/rules-reviewed/eap8/eap7/tests/hibernate6.windup.test.xml

```diff
@@ -0,0 +1,384 @@
+              <when>
+                  <not>
+                      <hint-exists message="The contents of `sql.ordering` were adapted and moved to `metamodel.mapping.ordering.ast`."/>
+                  </not>
+              </when>
+              <perform>
+                  <fail message="[hibernate6-tests-00140] Hint not found!"/>
+              </perform>
+          </rule>
+          <rule id="hibernate6-tests-00150">
+              <when>
+                  <not>
+                      <hint-exists message="Classes of the sql package that were previously used for building SQL, but aren’t needed anymore, were removed. The SQL generation is now fully handled through the `SqlAstTranslator` which a `Dialect` exposes a factory for."/>
+                  </not>
+              </when>
+              <perform>
+                  <fail message="[hibernate6-tests-00150] Hint not found!"/>
+              </perform>
+          </rule>
+          <rule id="hibernate6-tests-00160">
+              <when>
+                  <not>
+                      <hint-exists message="In Hibernate 6.0 the support for basic property mappings with multiple columns was removed. The only use case for that was when a `CompositeUserType` was in use, which was reworked to now work on top of components."/>
+                  </not>
+              </when>
+              <perform>
+                  <fail message="[hibernate6-tests-00160] Hint not found!"/>
+              </perform>
+          </rule>
+          <rule id="hibernate6-tests-00170">
+              <when>
+                  <not>
+                      <hint-exists message="The legacy Hibernate Criteria API which was deprecated back in Hibernate 5.x and removed in 6.0. Usually, all queries using the legacy API can be modeled with the JPA Criteria API. In some cases it is necessary to use the Hibernate JPA Criteria extensions."/>
+                  </not>
+              </when>
+              <perform>
+                  <fail message="[hibernate6-tests-00170] Hint not found!"/>
+              </perform>
+          </rule>
+          <rule id="hibernate6-tests-00180">
+              <when>
+                  <not>
+                      <hint-exists message="The Query#iterate\(\) method has been removed. See link for alternatives."/>
+                  </not>
+              </when>
+              <perform>
+                  <fail message="[hibernate6-tests-00180] Hint not found!"/>
+              </perform>
+          </rule>
+          <rule id="hibernate6-tests-00190">
+              <when>
+                  <not>
+                      <hint-exists message="Using `NativeQuery` to call SQL functions and procedures is no longer supported. `org.hibernate.procedure.ProcedureCall` or `jakarta.persistence.StoredProcedureQuery` should be used instead. `@NamedNativeQuery` references defining execution of procedure or functions should be migrated to use `@NamedStoredProcedureQuery` instead."/>
```

Comment:
I would enforce this check to use the `iterable-filter` with size = 2 to ensure both `javaclass` and `xmlfile` conditions from the rule are triggered.